AmplifyPlugins/API/AWSAPICategoryPluginFunctionalTests/GraphQLSyncBased/GraphQLSyncCustomPrimaryKeyTests.swift

```diff
@@ -0,0 +1,219 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import XCTest
+@testable import AWSAPICategoryPlugin
+@testable import Amplify
+@testable import AmplifyTestCommon
+@testable import AWSAPICategoryPluginTestCommon
+import AWSPluginsCore
+
+// swiftlint:disable cyclomatic_complexity
+class GraphQLSyncCustomPrimaryKeyTests: XCTestCase {
+
+    static let amplifyConfiguration = "GraphQLSyncBasedTests-amplifyconfiguration"
+
+    override func setUp() {
+        Amplify.reset()
+        let plugin = AWSAPIPlugin()
+
+        do {
+            try Amplify.add(plugin: plugin)
+
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(
+                forResource: GraphQLSyncBasedTests.amplifyConfiguration)
+            try Amplify.configure(amplifyConfig)
+
+            ModelRegistry.register(modelType: CustomerOrder.self)
+
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func tearDown() {
+        Amplify.reset()
+    }
+
+    /// Test for deletion of model with custom primary keys
+    ///
+    /// - Given: A customer order
+    /// - When:
+    ///    - The customer order is queried for, updated, deleted and queried for again
+    /// - Then:
+    ///    - The customer order should be deleted
+    ///
+    func testDelete() throws {
+
+        /// Create order
+        let customerOrder = CustomerOrder(orderId: UUID().uuidString, email: "test@abc.com")
+        guard let mutationSyncResult = createCustomerOrder(customerOrder: customerOrder),
+              let _ = mutationSyncResult.model.instance as? CustomerOrder else {
+            XCTFail("Failed to create customer order")
+            return
+        }
+
+        /// Query order
+        let querySuccess = expectation(description: "query successful")
+        let queryRequest = queryCustomerOrder(modelName: CustomerOrder.modelName,
+                                         byId: customerOrder.id,
+                                         orderId: customerOrder.orderId)
+        Amplify.API.query(request: queryRequest) { result in
```

Comment:
can you refactor like above? `createCustomerOrder`, `queryCustomerOrder`, `deleteCustomerOrder`, `updateCustomerOrder`, etc